src/transformers/pipelines.py

```diff
@@ -553,6 +553,232 @@ def __call__(self, *args, **kwargs):
+        self.xlm_language = xlm_language
+        self.length = length
+        self.temperature = temperature
+        self.top_k = top_k
+        self.top_p = top_p
+        self.repetition_penalty = repetition_penalty
+        self.do_sample = do_sample
+        self.num_return_sequences = num_return_sequences
+        self.model_type = self.model.config.to_dict()["model_type"]
+        self.preprocessing_functions = {
+            "ctrl": self.prepare_ctrl_input,
+            "xlm": self.prepare_xlm_input,
+            "xlnet": self.prepare_xlnet_input,
+            "transfo-xl": self.prepare_transfoxl_input,
+        }
+
+    def __call__(self, *texts, **kwargs):
+        inputs = self._args_parser(*texts, **kwargs)
+
+        all_generated_sequences = []
+        for i, prompt_text in enumerate(inputs):
+            # Manage correct placement of the tensors
+            with self.device_placement():
+                tokens = self.prepare_input(prompt_text)
+                if self.framework == "pt":
+                    with torch.no_grad():
+                        tokens = tokens.to(self.device)
+
+                        output_sequences = self.model.generate(
+                            input_ids=tokens,  # BS x SL
+                            max_length=self.length + len(tokens.squeeze()),  # SL
+                            temperature=self.temperature,
+                            top_k=self.top_k,
+                            top_p=self.top_p,
+                            repetition_penalty=self.repetition_penalty,
+                            do_sample=self.do_sample,
+                            num_return_sequences=self.num_return_sequences,
+                        )
+
+                else:
+                    output_sequences = self.model.generate(
+                        input_ids=tokens,  # BS x SL
+                        max_length=self.length + len(tokens.squeeze()),  # SL
+                        temperature=self.temperature,
+                        top_k=self.top_k,
+                        top_p=self.top_p,
+                        repetition_penalty=self.repetition_penalty,
+                        do_sample=self.do_sample,
+                        num_return_sequences=self.num_return_sequences,
+                    )
+
+            # Remove the batch dimension when returning multiple sequences
+            if len(output_sequences.shape) > 2:
+                output_sequences.squeeze_()
+
+            generated_sequences = []
+
+            for generated_sequence_idx, generated_sequence in enumerate(output_sequences):
+                generated_sequence = generated_sequence.tolist()
+
+                # Decode text
+                text = self.tokenizer.decode(generated_sequence, clean_up_tokenization_spaces=True)
+
+                # Remove all text after the stop token
+                text = text[: text.find(self.stop_token) if self.stop_token else None]
```

Comment:
I don't think we need this either no? Generation automatically stops at EOS so this should not be needed.

Reply:
got it makes sense